pkg/agent/util/net_linux_test.go

```diff
@@ -0,0 +1,853 @@
+	return &netlink.LinkAttrs{
+		Index:        l.index,
+		Name:         l.name,
+		MasterIndex:  l.masterIndex,
+		HardwareAddr: l.hardwareAddr,
+	}
+}
+
+func (l mockLink) Type() string {
+	return "mock"
+}
+
+type mockNetNS struct {
+	file   string
+	closed bool
+}
+
+func (ns *mockNetNS) Do(toRun func(ns.NetNS) error) error {
+	return toRun(&mockNetNS{})
+}
+
+func (ns *mockNetNS) Set() error {
+	return nil
+}
+
+func (ns *mockNetNS) Path() string {
+	return ns.file
+}
+
+func (ns *mockNetNS) Fd() uintptr {
+	return 0
+}
+
+func (ns *mockNetNS) Close() error {
+	ns.closed = true
+	return nil
+}
+
+func TestGetNSPeerDevBridge(t *testing.T) {
+	tests := []struct {
+		name                string
+		getNSErr            error
+		getVethPeerErr      error
+		testNetInterface    *mockNetInterface
+		testNetInterfaceErr error
+		peerLink            netlink.Link
+		netlinkErr          error
+		bridgeLink          netlink.Bridge
+		bridgeErr           error
+		wantErr             error
+		wantNetInterface    bool
+		wantName            string
+	}{
+		{
+			name: "Get Bridge",
+			peerLink: mockLink{
+				masterIndex: 1,
+			},
+			bridgeLink: netlink.Bridge{
+				LinkAttrs: netlink.LinkAttrs{
+					Name: "test-b0",
+				},
+			},
+			wantErr:          nil,
+			wantName:         "test-b0",
+			wantNetInterface: true,
+		},
+		{
+			name:             "Get NS Err",
+			getNSErr:         fmt.Errorf("invalid"),
+			wantErr:          fmt.Errorf("failed to get NS for path"),
+			wantName:         "",
+			wantNetInterface: false,
+		},
+		{
+			name:             "Get V-eth Peer Err",
+			getVethPeerErr:   fmt.Errorf("invalid"),
+			wantErr:          fmt.Errorf("failed to get peer idx for dev"),
+			wantName:         "",
+			wantNetInterface: false,
+		},
+		{
+			name:                "Get Interface Err",
+			testNetInterfaceErr: fmt.Errorf("invalid"),
+			wantErr:             fmt.Errorf("failed to get interface for idx"),
+			wantName:            "",
+			wantNetInterface:    false,
+		},
+		{
+			name:             "Get Link Err",
+			netlinkErr:       fmt.Errorf("invalid"),
+			wantErr:          fmt.Errorf("failed to get link for idx"),
+			wantName:         "",
+			wantNetInterface: false,
+		},
+		{
+			name: "Not Attached to Bridge",
+			peerLink: mockLink{
+				masterIndex: -1,
+			},
+			wantErr:          nil,
+			wantName:         "",
+			wantNetInterface: true,
+		},
+		{
+			name: "Get Bridge Err",
+			peerLink: mockLink{
+				masterIndex: 1,
+			},
+			bridgeErr:        fmt.Errorf("invalid"),
+			wantErr:          fmt.Errorf("failed to get master link for dev"),
+			wantName:         "",
+			wantNetInterface: false,
+		},
+		{
+			name: "Master Link not Bridge",
+			peerLink: mockLink{
+				masterIndex: 2,
+			},
+			wantErr:          nil,
+			wantName:         "",
+			wantNetInterface: true,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			getNS = func(nspath string) (ns.NetNS, error) {
```

Comment:
Please remember to restore the overwritten functions after test is completed, this is suitable for all the test files in this patch.

Reply:
Checked all the occurrences, also grouped the restore functions in one `defer`.

Reply:
Refactored all the mock functions, during development, discovered that **defer of the return function** is executed outside the` testcases for loop`. Because the deferred functions are the same, this should not affect the result, but if we want to improve this across all UT in the codebase, we might use anonymous functions.